common/scala/src/main/scala/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -135,9 +151,21 @@ class KafkaConsumerConnector(
   }
 
   private def recreateConsumer(): Unit = {
-    val oldConsumer = consumer
-    oldConsumer.close()
-    logging.info(this, s"old consumer closed")
+    logging.info(this, s"recreating consumer for '$topic'")
+    try {
+      val oldConsumer = consumer
+      oldConsumer.close()
+    } catch {
+      // According to documentation, the consumer is force closed if it cannot be closed gracefully.
+      // See https://kafka.apache.org/11/javadoc/index.html?org/apache/kafka/clients/consumer/KafkaConsumer.html
+      //
+      // For the moment, we are ignoring 'InterruptException' which indicates that the thread has been
+      // interrupted while running 'close()'.
+      //
```

Comment:
I actually meant that we don't have special handling of `InterruptException`. It may be needed to re-try the `close()` if this exception occurs.

I will update the comment.